src/marqo/index.py

```diff
@@ -228,7 +228,7 @@ def get_status(self):
         else:
             raise UnsupportedOperationError("This operation is only supported for Marqo Cloud")
 
-    def search(self, q: Union[str, dict], searchable_attributes: Optional[List[str]] = None,
+    def search(self, q: Optional[Union[str, dict]] = None, searchable_attributes: Optional[List[str]] = None,
                limit: int = 10, offset: int = 0, search_method: Union[SearchMethods.TENSOR, str] = SearchMethods.TENSOR,
```

Comment:
Just to confirm we set `q = None` as the default value instead of asking for an explicitly `None` value in the API design

Reply:
Yes, this design is meant so users can search without specifying the q parameter at all. For example:
```
self.client.index(test_index_name).search(
            context={"tensor": [{"vector": custom_vector_3, "weight": 1}]}  # no text query
        )
```